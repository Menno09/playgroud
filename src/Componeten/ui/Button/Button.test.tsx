import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import React from 'react';

import '@testing-library/jest-dom/extend-expect';

import Props from './Button.props';
import {
  Black,
  Green,
  LightPink,
  Pink,
  Primary,
  PrimaryWithLink,
  Secondary,
} from './Button.stories';

describe('Button component tests', () => {
  it('renders the Button in the primary state', () => {
    const props: Props = Primary.args as Props;
    const onClick = jest.fn();

    const { container } = render(
      <div data-testid="Button">
        <Primary {...props} onClick={onClick} />
      </div>,
    );

    const button = container.querySelectorAll('button')[0]; // Example usage of container, to bypass eslint issue
    userEvent.click(button);

    expect(button).toBeInTheDocument();
    expect(button).toMatchSnapshot();
    expect(onClick).toHaveBeenCalled();
  });

  it('renders the Button in the secondary state', () => {
    const props: Props = Secondary.args as Props;
    const onClick = jest.fn();

    const { container } = render(
      <div data-testid="Button">
        <Secondary {...props} onClick={onClick} />
      </div>,
    );

    const button = container.querySelectorAll('button')[0]; // Example usage of container, to bypass eslint issue
    userEvent.click(button);

    expect(button).toBeInTheDocument();
    expect(button).toMatchSnapshot();
    expect(onClick).toHaveBeenCalled();
  });

  it('renders the Button in the green state', () => {
    const props: Props = Green.args as Props;
    const onClick = jest.fn();

    const { container } = render(
      <div data-testid="Button">
        <Green {...props} onClick={onClick} />
      </div>,
    );

    const button = container.querySelectorAll('button')[0]; // Example usage of container, to bypass eslint issue
    userEvent.click(button);

    expect(button).toBeInTheDocument();
    expect(button).toMatchSnapshot();
    expect(onClick).toHaveBeenCalled();
  });

  it('renders the Button in the pink state', () => {
    const props: Props = Pink.args as Props;
    const onClick = jest.fn();

    const { container } = render(
      <div data-testid="Button">
        <Pink {...props} onClick={onClick} />
      </div>,
    );

    const button = container.querySelectorAll('button')[0]; // Example usage of container, to bypass eslint issue
    userEvent.click(button);

    expect(button).toBeInTheDocument();
    expect(button).toMatchSnapshot();
    expect(onClick).toHaveBeenCalled();
  });

  it('renders the Button in the light pink state', () => {
    const props: Props = LightPink.args as Props;
    const onClick = jest.fn();

    const { container } = render(
      <div data-testid="Button">
        <LightPink {...props} onClick={onClick} />
      </div>,
    );

    const button = container.querySelectorAll('button')[0]; // Example usage of container, to bypass eslint issue
    userEvent.click(button);

    expect(button).toBeInTheDocument();
    expect(button).toMatchSnapshot();
    expect(onClick).toHaveBeenCalled();
  });

  it('renders the Button in the black state', () => {
    const props: Props = Black.args as Props;
    const onClick = jest.fn();

    const { container } = render(
      <div data-testid="Button">
        <Black {...props} onClick={onClick} />
      </div>,
    );

    const button = container.querySelectorAll('button')[0]; // Example usage of container, to bypass eslint issue
    userEvent.click(button);

    expect(button).toBeInTheDocument();
    expect(button).toMatchSnapshot();
    expect(onClick).toHaveBeenCalled();
  });

  it('renders the Button in the primary state with a link', () => {
    const props: Props = PrimaryWithLink.args as Props;
    const onClick = jest.fn();

    const { container } = render(
      <div data-testid="Button">
        <PrimaryWithLink {...props} onClick={onClick} />
      </div>,
    );

    const button = container.querySelectorAll('a')[0];

    expect(button).toBeInTheDocument();
    expect(button).toMatchSnapshot();
    expect(button).toHaveAttribute('href', 'https://google.com');
  });
});
