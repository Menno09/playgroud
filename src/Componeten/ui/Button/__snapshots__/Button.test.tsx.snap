// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component tests renders the Button in the black state 1`] = `
<button
  class="Button btn Button--black"
  type="button"
>
  <div
    class="Button__ChildContainer"
  >
    Klik hier!
  </div>
</button>
`;

exports[`Button component tests renders the Button in the green state 1`] = `
<button
  class="Button btn Button--green"
  type="button"
>
  <div
    class="Button__ChildContainer"
  >
    Klik hier!
  </div>
</button>
`;

exports[`Button component tests renders the Button in the light pink state 1`] = `
<button
  class="Button btn Button--pink"
  type="button"
>
  <div
    class="Button__ChildContainer"
  >
    Klik hier!
  </div>
</button>
`;

exports[`Button component tests renders the Button in the pink state 1`] = `
<button
  class="Button btn Button--pink"
  type="button"
>
  <div
    class="Button__ChildContainer"
  >
    Klik hier!
  </div>
</button>
`;

exports[`Button component tests renders the Button in the primary state 1`] = `
<button
  class="Button btn Button--primary"
  type="button"
>
  <div
    class="Button__ChildContainer"
  >
    Klik hier!
  </div>
</button>
`;

exports[`Button component tests renders the Button in the primary state with a link 1`] = `
<a
  class="Button btn Button--primary"
  href="https://google.com"
>
  Klik hier!
</a>
`;

exports[`Button component tests renders the Button in the secondary state 1`] = `
<button
  class="Button btn Button--secondary"
  type="button"
>
  <div
    class="Button__ChildContainer"
  >
    Klik hier!
  </div>
</button>
`;
